### PR TITLE
feat: django-framework extension

### PR DESCRIPTION
**Pull request by jdkandersson**
As discussed with @sergiusens we are proposing to keep the django extension in `latest/edge` in experimental and not pick it for candidate releases until the extension is no longer experimental.

### CODE REVIEW

File: charmcraft/templates/init-django-framework/src/charm.py.j2 [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import paas_app_charmer.django
+
+logger = logging.getLogger(__name__)
+
+
+class DjangoCharm(paas_app_charmer.django.Charm):
+    """Django Charm service."""
+
+    def __init__(self, *args: typing.Any) -> None:
```
[the remark this answers is not in the document]
Comment (jdkandersson) [reply charmcraft/templates/init-django-framework/src/charm.py.j2:20]: It is a product requirement that we leave a path open for developers to take over aspects of charming, such as observing their own events and so on. So this is to give developers that want to customise an entry point

File: charmcraft/templates/init-django-framework/src/charm.py.j2 [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+# Copyright {{ year }} {{ author }}
+# See LICENSE file for licensing details.
+
+"""Flask Charm entrypoint."""
```
[the remark this answers is not in the document]
Comment (jdkandersson) [reply charmcraft/templates/init-django-framework/src/charm.py.j2]: Thanks for catching, should be fixed now

**Review by tigarmo — APPROVED** [review]

**Review by jdkandersson — COMMENTED** [review]
self-review done

**Comment by jdkandersson** [comment]
Ok, I think I added `fetch-libs` support, might need to wait for a fix of #1671

**Review by jdkandersson — COMMENTED** [review]
self review done

**Review by lengau — APPROVED** [review]

File: charmcraft/extensions/__init__.py [hunk 3]
```diff
@@ -30,6 +30,7 @@
 register("flask-framework", FlaskFramework)
```
Comment (sergiusens) [inline charmcraft/extensions/__init__.py:30]: what about this line here? We seem to be double registering

**Pull request merged**